ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -2885,8 +2886,13 @@ private RelNode genTableLogicalPlan(String tableAlias, QB qb) throws SemanticExc
         // Virtual Cols
 
         // 3.1 Add Column info for non partion cols (Object Inspector fields)
-        StructObjectInspector rowObjectInspector = (StructObjectInspector) tabMetaData.getDeserializer()
+        final Deserializer deserializer = tabMetaData.getDeserializer();
+        StructObjectInspector rowObjectInspector = (StructObjectInspector) deserializer
             .getObjectInspector();
+
+        // Special handling for iceberg tables which set specific configurations during initialisation of Serde.
```

Comment:
No need to mention Iceberg here, the interface documentation should be clear instead what this method is doing.